proposals/0652-import-shadowing.rst

```diff
@@ -265,29 +265,31 @@ something like
 then the ``foo`` exported by ``A`` should be the one defined in
 ``A``'s top-level.
 
-When modules are reexported wholesale, shadowing doesn't come into
-play, and so we keep the behaviour without this extension: the form
-``module M`` names the set of all entities that are in scope with both
-an unqualified name ``e`` and a qualified name ``M.e``. Example:
+When modules are reexported wholesale,
+`Section 5.2.5 of the Haskell 2010 report
+<https://www.haskell.org/onlinereport/haskell2010/haskellch5.html#x11-1000005.2>`_
+prescribes that only entities which are in scope from ``M``
+unqualified are reexported. Example:
```

Comment:
I think it is unhelpful to rephrase what the report says, and especially to do so in an ambiguous way.   How about this instead.

```
In an export list, `Section 5.2.5 of the Haskell 2010 report
<https://www.haskell.org/onlinereport/haskell2010/haskellch5.html#x11-1000005.2>`_ specifies that the form
``module M`` names the set of all entities that are in scope with both
an unqualified name ``e`` and a qualified name ``M.e``.  So in this example

module A( module M ) where
   import M(foo, wombat)
   import N(foo)
   foo = True

the `module M` exports `M.foo` because that entity is in scope both as `M.foo` and with unqualified name `foo`.  The fact that `N.foo` and `A.foo` are *also* in scope with unqualified name `foo` does not matter.

With `ImportShadowing`, however, the local definition of `A.foo` *completely hides* the unqualified imports of `M.foo` and `N.foo`; so now `M.foo` is no longer in scope with unqualified name `foo`; so the `module M` export list exports only `M.wombat`.
```